File: challenges/5-cobaias/javascript/codinghobbit/desafio5.js
const readline = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});

readline.question(
  "Insira os dados de pesquisa. Ex '10 10 C 6 R 15 S ...' ",
  (researchData) => {
    const totalInputTests = researchData.split(" ")[0];
    const data = researchData.split(" ").slice(1);
    if (data.length !== totalInputTests * 2) {
      console.log("Insira a quantidade correta de dados");
      readline.close();
      return
    }
    const dataCounters = { total: 0, rabbits: 0, rats: 0, frogs: 0 };

    data.forEach((item, index) => {
      const animal = data[index + 1];
      if (index % 2 === 0) {
        if(animal.toUpperCase() === 'C' || animal.toUpperCase() === 'R'|| animal.toUpperCase() === 'S'){
          dataCounters.total += parseInt(item);
          animal.toUpperCase() === "C"
          ? (dataCounters.rabbits += parseInt(item))
          : animal.toUpperCase() === "R"
          ? (dataCounters.rats += parseInt(item))
          : animal.toUpperCase() === "S"
          ? (dataCounters.frogs += parseInt(item))
          : null;
        }
      }
    });
    console.log(`
    Total: ${dataCounters.total} cobaias
    Total de coelhos: ${dataCounters.rabbits}
    Total de ratos: ${dataCounters.rats}
    Total de sapos: ${dataCounters.frogs}
    Percentual de coelhos: ${(
      (dataCounters.rabbits / dataCounters.total) *
      100
    ).toFixed(2)}%
    Percentual de ratos: ${(
      (dataCounters.rats / dataCounters.total) *
      100
    ).toFixed(2)}%
    Percentual de sapos: ${(
      (dataCounters.frogs / dataCounters.total) *
      100
    ).toFixed(2)}%`);

    readline.close();
  }
);

File: challenges/5-cobaias/python/scush989898/desafio-5.py
test_cases = int(input())
animals_list = {"S": 0, "R": 0, "C": 0}

sum_animals = []

for i in range(test_cases):
    current = str(input()).strip().upper()
    sum_animals.append(current.split(" "))


def add_to_current_animal(letter: str, value: int) -> None:
    for k, v in animals_list.items():
        if k == letter:
            animals_list[k] = v + value


def get_percentage(animal: int, total: int) -> str:
    percentage = animal / total * 100
    return "%.2f" % percentage + " %"


for item in sum_animals:
    value = int(item[0])
    animal = item[1]
    add_to_current_animal(animal, value)

total = sum(list(animals_list.values()))

coelhos = animals_list["C"]
ratos = animals_list["R"]
sapos = animals_list["S"]

print(f"Total: {total} cobaias")
print(f"Total de coelhos: {coelhos}")
print(f"Total de ratos: {ratos}")
print(f"Total de sapos: {sapos}")
print(f"Percentual de coelhos: {get_percentage(coelhos, total)}")
print(f"Percentual de ratos: {get_percentage(ratos, total)}")
print(f"Percentual de sapos: {get_percentage(sapos, total)}")
